crates/fluvio-connector-package/src/secret.rs

```diff
@@ -453,16 +493,60 @@ config:
         name: secret3
     - secret:
         name: secret4
+  string_interpolation_in_quotes: "My name is ${{ secrets.secret5 }}"
+  string_interpolation_without_quotes: ${{ secrets.secret6 }}
+  string_interpolation_without_spaces: ${{secrets.secret7}}
+  list_interpolation:
+    - ${{ secrets.secret8 }}
+    - ${{ secrets.secret9 }}
+  key: value
+  num: 100
+  bool: true
+  obj:
+    key: value
 "#;
 
         //when
-        let secrets = detect_secrets_from_str(raw_config).expect("invalid raw config string");
+        let secrets: HashSet<String> =
+            detect_secrets_from_str(raw_config).expect("invalid raw config string");
 
         //then
-        assert_eq!(secrets.len(), 4);
+
+        assert_eq!(secrets.len(), 9);
         assert!(secrets.get("secret1").is_some());
         assert!(secrets.get("secret2").is_some());
         assert!(secrets.get("secret3").is_some());
         assert!(secrets.get("secret4").is_some());
+        assert!(secrets.get("secret5").is_some());
+        assert!(secrets.get("secret6").is_some());
+        assert!(secrets.get("secret7").is_some());
```

Comment:
shouldn't this be none? since there is no secret7?

Reply:
There is one reference to secret7: 
```
string_interpolation_without_spaces: ${{secrets.secret7}}
```